soql-standalone-parser/src/main/scala/com/socrata/soql/parsing/RecursiveDescentHintParser.scala

```diff
@@ -0,0 +1,54 @@
+package com.socrata.soql.parsing
+
+import com.socrata.soql.ast.{Hint, Materialized}
+import com.socrata.soql.parsing.RecursiveDescentParser.{AnIdentifier, ParseResult, Reader}
+import com.socrata.soql.tokens.{COMMA, HINT, LPAREN, RPAREN}
+
+import scala.annotation.tailrec
+import scala.util.parsing.input.Position
+
+trait RecursiveDescentHintParser { this: RecursiveDescentParser =>
+
+  protected final def hints(reader: Reader): ParseResult[Seq[Hint]] = {
+    reader.first match {
+      case HINT() =>
+        reader.rest.first match {
+          case LPAREN() =>
+            parseHintList(reader.rest.rest)
+        }
```

Comment:
This will throw a MatchError if it sees a HINT followed by not-an-LPAREN.  You'll want a `case _ => ParseResult(reader, Seq.empty)` here too, to not consume the branch (or `fail` if you do want to keep `hint` a reserved word)